File: app/src/main/java/com/youngwu/kotlin/activity/MainActivity.java
package com.youngwu.kotlin.activity;

import android.os.Bundle;
import android.view.View;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.AppCompatButton;

import com.youngwu.kotlin.demo.R;

/**
 * @author yinyangwu
 * @date 2020/5/13 14:23
 * description:
 */
public class MainActivity extends AppCompatActivity implements View.OnClickListener {
    private AppCompatButton btnFinish;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        btnFinish = findViewById(R.id.btnFinish);
        btnFinish.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        finish();
    }
}

File: app/src/main/java/com/youngwu/kotlin/activity/MainActivity.kt
package com.youngwu.kotlin.activity

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.youngwu.kotlin.demo.R
import kotlinx.android.synthetic.main.activity_main.*

/**
 * @author yinyangwu
 * @date 2020/5/13 14:23
 * description:
 */
class MainActivity : AppCompatActivity(), View.OnClickListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnFinish.setOnClickListener(this)
    }

    override fun onClick(v: View) {
        finish()
    }
}